### ISSUE TITLE
<b-modal> header is not read for accessibility with JAWS

### ISSUE THREAD

**Issue by jamart2005** [bootstrap-vue/bootstrap-vue#3712]
### Description
b-modal accessibility with Chrome/JAWS doesn't recognize the modal's header as its label even when you pass the id prop

### Steps to reproduce the bug

1. Go to https://bootstrap-vue.js.org/docs/components/modal/
2. Click on "Launch demo modal"
3. Modal was opened but the header is not read, even after tabbing to the close button

### Expected behavior
The modal header should be considered as dialog's label and be read when modal is open or atleast when tabbed to the close button (this works in NVDA once focus reach the close button)

**Environment:**
- OS: Windows
- Browser: Chrome
- Version: 74
- Screenreader: JAWS 2019, NVDA 2017.2

### Additional context
It seems to be having the aria-labelledby in the role="dialog" DIV instead of in role="document" DIV enables it to read on both JAWS and NVDA when modal is opened.  If its added in the the role="dialog" DIV and role="document" DIV still have aria-labelledby, it would read twice with NVDA once focus goes to close button when

**Comment by tmorehouse** [comment]
@jamart2005 thanks for finding this out.

PR #3713 hopefully will address this issue.

Could you try testing it out at https://deploy-preview-3713--bootstrap-vue.netlify.com/

**Comment by jamart2005** [comment]
@tmorehouse thanks for the quick turnaround!

It does work now per my testing with JAWS and NVDA. It reads header and body once modal opens

Minor issue and would be nice if its handled.  since it reads the whole header now, it also announces the close button when modal opens which sounds unnecessary. I'm not that familiar under the hood of bootstrapVue but I was thinking of possible solution is have aria-labelledby point to modal-title if no modal-header slot is passed else just point to modal-header.

**Comment by tmorehouse** [comment]
That might be possible... maybe using a computed prop to figure out which to point to.

**Comment by tmorehouse** [comment]
Bootstrap v2.0.0-rc.27 has been released
